packages/iov-bns/src/decode.spec.ts

```diff
@@ -216,6 +179,75 @@ describe("Decode", () => {
     });
   });
 
+  describe("decodeJsonAmount", () => {
+    it("can decode strings ", () => {
+      const decoded = decodeJsonAmount(`"3.123456789 ASH"`);
+      expect(decoded).toEqual({
+        quantity: "3123456789",
+        fractionalDigits: 9,
+        tokenTicker: "ASH" as TokenTicker,
+      });
+
+      const decoded2 = decodeJsonAmount(`"4IOV"`);
+      expect(decoded2).toEqual({
+        quantity: "4000000000",
+        fractionalDigits: 9,
+        tokenTicker: "IOV" as TokenTicker,
+      });
+
+      const decoded3 = decodeJsonAmount(`"0.0001   CASH"`);
+      expect(decoded3).toEqual({
+        quantity: "100000",
+        fractionalDigits: 9,
+        tokenTicker: "CASH" as TokenTicker,
+      });
+    });
+
+    it("rejects invalid strings ", () => {
+      // invalid format
+      expect(() => decodeJsonAmount(`"1,23 ASH"`)).toThrowError();
+      // leading spaces
+      expect(() => decodeJsonAmount(`"   1 ASH"`)).toThrowError();
+      // no whole numbers
+      expect(() => decodeJsonAmount(`".0001   CASH"`)).toThrowError();
+      // just number
+      expect(() => decodeJsonAmount(`"42.13"`)).toThrowError();
+      // just ticker
+      expect(() => decodeJsonAmount(`"FOO"`)).toThrowError();
+      // wrong order
+      expect(() => decodeJsonAmount(`"ASH 22"`)).toThrowError();
+      // ticker bad size
+      expect(() => decodeJsonAmount(`"0.01 A"`)).toThrowError();
+      // ticker bad size
+      expect(() => decodeJsonAmount(`"0.01 SUPERLONG"`)).toThrowError();
+    });
```

Comment:
Can we test more than 9 fractional positions? e.g. `0.0123456789` This cannot be represented as a weave amount.

Reply:
Good point